File: backend/local-api-module/src/main/java/org/vyatsu/localApiModule/entity/item/Item.java
package org.vyatsu.localApiModule.entity.item;

import jakarta.persistence.*;
import lombok.*;
import org.vyatsu.localApiModule.entity.contractor.Contractor;
import org.vyatsu.localApiModule.entity.contractor.WalletTransaction;
import org.vyatsu.localApiModule.entity.purchase.Ad;
import org.vyatsu.localApiModule.entity.purchase.PurchaseItem;
import org.vyatsu.localApiModule.entity.user.User;

import java.time.LocalDateTime;
import java.util.LinkedHashSet;
import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
@Entity
@Table(name = "items")
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "first_price")
    private Double firstPrice;

    @ManyToOne
    @JoinColumn(name = "id_user", nullable = false)
    private User user;

    @Column(name = "category")
    private String category;

    @Column(name = "vendor_code")
    private String vendorCode;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;

    @Column(name = "is_draft", nullable = false)
    private Boolean isDraft;

    @Column(name = "created_at", columnDefinition = "TIMESTAMP NOT NULL")
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "item")
    private Set<Order> orders = new LinkedHashSet<>();

    @OneToMany(mappedBy = "item")
    private Set<PriceHistory> priceHistory = new LinkedHashSet<>();

    @OneToMany(mappedBy = "item")
    private Set<Ad> ads = new LinkedHashSet<>();

    @OneToMany(mappedBy = "item")
    private Set<PurchaseItem> purchaseItems = new LinkedHashSet<>();

    @OneToMany(mappedBy = "item")
    private Set<WalletTransaction> walletTransactions = new LinkedHashSet<>();

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "contractor_item",
            joinColumns = @JoinColumn(name = "item_id", nullable = false),
            inverseJoinColumns = @JoinColumn(name = "contractor_id", nullable = false)
    )
    private Set<Contractor> contractors;
}

File: app/src/main/java/com/bytebusters/android/mpsystem/entity/item/Item.kt
package com.bytebusters.android.mpsystem.entity.item

import com.bytebusters.android.mpsystem.entity.purchase.Ad
import com.bytebusters.android.mpsystem.entity.purchase.PurchaseItem
import com.bytebusters.android.mpsystem.entity.user.User
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.time.LocalDateTime

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "items")
class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private val id: Long? = null

    @Column(name = "title", nullable = false)
    private val title: String? = null

    @Column(name = "description", length = 500)
    private val description: String? = null

    @Column(name = "first_price")
    private val firstPrice: Double? = null

    @ManyToOne
    @JoinColumn(name = "id_user", nullable = false)
    private val user: User? = null

    @Column(name = "category")
    private val category: String? = null

    @Column(name = "vendor_code")
    private val vendorCode: String? = null

    @Column(name = "is_active", nullable = false)
    private val isActive: Boolean? = null

    @Column(name = "is_draft", nullable = false)
    private val isDraft: Boolean? = null

    @Column(name = "created_at", columnDefinition = "TIMESTAMP NOT NULL")
    private val createdAt: LocalDateTime? = null

    @OneToMany(mappedBy = "item")
    private val orders: Set<Order> = LinkedHashSet()

    @OneToMany(mappedBy = "item")
    private val priceHistory: Set<PriceHistory> = LinkedHashSet()

    @OneToMany(mappedBy = "item")
    private val ads: Set<Ad> = LinkedHashSet<Ad>()

    @OneToMany(mappedBy = "item")
    private val purchaseItems: Set<PurchaseItem> = LinkedHashSet<PurchaseItem>()
}
